popcorn/popcorn.rb

```diff
@@ -0,0 +1,60 @@
+require 'set'
+
+class Popcorn
+    Words = %w(cat dog pop corn popcorn crop corp)
+
+    class Node
+        attr_reader :connections, :char
+        attr_accessor :used
+
+        def initialize(char, connections)
+            @char = char
+            @used = false
+            @connections = Set.new(connections)
```

Comment:
Usually it's not a node's responsibility to know whether it's `used?` or what its `connections` are, because you can have cycles, edges from a node to itself, multiple edges with weights and so on.
You may also want to traverse the same graph using different algorithms and maybe in parallel - so it's a graph visitor's responsibility to track what's visited.